src/storage-client/src/sink.rs

```diff
@@ -634,7 +634,7 @@ pub async fn determine_sink_resume_upper(
         // the high water mark, and therefore `last_timestamp` contains the
         // most recent timestamp for the sink under consideration.
         Ok(last_upper)
-    }).await.unwrap().check_ssh_status(&ctx)
+    }).abort_on_drop().await.unwrap().check_ssh_status(&ctx)
```

Comment:
Does this have any effect? The docs on `spawn_blocking` have this to say:

> Closures spawned using spawn_blocking cannot be cancelled abruptly; there is no standard low level API to cause a thread to stop running. However, a useful pattern is to pass some form of “cancellation token” into the thread. This could be an [AtomicBool](https://doc.rust-lang.org/nightly/core/sync/atomic/struct.AtomicBool.html) that the task checks periodically. Another approach is to have the thread primarily read or write from a channel, and to exit when the channel closes; assuming the other side of the channel is dropped when cancellation occurs, this will cause the blocking task thread to exit soon after as well.

> When you shut down the executor, it will wait indefinitely for all blocking operations to finish. You can use [shutdown_timeout](https://docs.rs/tokio/latest/tokio/runtime/struct.Runtime.html#method.shutdown_timeout) to stop waiting for them after a certain timeout. Be aware that this will still not cancel the tasks — they are simply allowed to keep running after the method returns. It is possible for a blocking task to be cancelled if it has not yet started running, but this is not guaranteed.

Reply:
Oh I though this was a normal tokio task; yeah I don't think this does anything

Reply:
duh, thanks for catching this! I wired up a cancellation token as the docs suggest. But also, why do blocking tasks have an `.abort()` method that does nothing? What is this, Javascript?

Reply:
@petrosagg its because they share the `JoinHandle` type, and fixing that is a breaking change :(